pkg/otlp/metrics/metrics_remapping.go

```diff
@@ -0,0 +1,193 @@
+	case "container.memory.hierarchical_memory_limit":
+		copyMetric(all, m, "container.memory.limit", 1)
+	case "container.memory.usage.limit":
+		copyMetric(all, m, "container.memory.soft_limit", 1)
+	case "container.memory.total_cache":
+		copyMetric(all, m, "container.memory.cache", 1)
+	case "container.memory.total_swap":
+		copyMetric(all, m, "container.memory.swap", 1)
+	case "container.blockio.io_service_bytes_recursive":
+		copyMetric(all, m, "container.io.write", 1, kv{"operation", "write"})
+		copyMetric(all, m, "container.io.read", 1, kv{"operation", "read"})
+	case "container.blockio.io_serviced_recursive":
+		copyMetric(all, m, "container.io.write.operations", 1, kv{"operation", "write"})
+		copyMetric(all, m, "container.io.read.operations", 1, kv{"operation", "read"})
+	case "container.network.io.usage.tx_bytes":
+		copyMetric(all, m, "container.net.sent", 1)
+	case "container.network.io.usage.tx_packets":
+		copyMetric(all, m, "container.net.sent.packets", 1)
+	case "container.network.io.usage.rx_bytes":
+		copyMetric(all, m, "container.net.rcvd", 1)
+	case "container.network.io.usage.rx_packets":
+		copyMetric(all, m, "container.net.rcvd.packets", 1)
+	}
+}
+
+// kv represents a key/value pair.
+type kv struct{ K, V string }
+
+// copyMetric copies metric m to dest. The new metric's name will be newname, and all of its datapoints will
+// be divided by div. If filter is provided, only the data points that have *either* of the specified string
+// attributes will be copied over. If the filtering results in no datapoints, no new metric is added to dest.
+//
+// Please note that copyMetric is restricted to the metric types Sum and Gauge.
+func copyMetric(dest pmetric.MetricSlice, m pmetric.Metric, newname string, div float64, filter ...kv) (pmetric.Metric, bool) {
+	newm := pmetric.NewMetric()
+	m.CopyTo(newm)
+	newm.SetName(newname)
+	var dps pmetric.NumberDataPointSlice
+	switch newm.Type() {
+	case pmetric.MetricTypeGauge:
+		dps = newm.Gauge().DataPoints()
+	case pmetric.MetricTypeSum:
+		dps = newm.Sum().DataPoints()
+	default:
+		// invalid metric type
+		return newm, false
+	}
+	dps.RemoveIf(func(dp pmetric.NumberDataPoint) bool {
+		if !hasAny(dp, filter...) {
+			return true
+		}
+		switch dp.ValueType() {
+		case pmetric.NumberDataPointValueTypeInt:
+			dp.SetIntValue(dp.IntValue() / int64(div))
+		case pmetric.NumberDataPointValueTypeDouble:
+			dp.SetDoubleValue(dp.DoubleValue() / div)
+		}
+		return false
+	})
+	if dps.Len() > 0 {
+		// if we have datapoints, copy it
+		addm := dest.AppendEmpty()
+		newm.CopyTo(addm)
+		return addm, true
+	}
+	return newm, false
+}
+
+// hasAny reports whether point has any of the given string tags.
+// If no tags are provided it returns true.
+func hasAny(point pmetric.NumberDataPoint, tags ...kv) bool {
+	if len(tags) == 0 {
+		return true
+	}
+	attr := point.Attributes()
+	for _, tag := range tags {
+		v, ok := attr.Get(tag.K)
+		if !ok {
+			continue
+		}
+		if v.Str() == tag.V {
```

Comment:
This panics if `v.Type() != pcommon.ValueTypeStr`. I think we should check for this first (needs a new import):
```suggestion
		if v.Type() == pcommon.ValueTypeStr  && v.Str() == tag.V {
```

Reply:
Ah, right, my bad, I got confused with how `pmetric.Metric` works. Marking this as solved